src/client/public/scripts/components/base-element.ts

```diff
@@ -1,17 +1,44 @@
 import {render} from '../../../../../node_modules/lit-html/lib/lit-extended.js';
 import {TemplateResult} from '../../../../../node_modules/lit-html/lit-html.js';
 
+type Debouncer = (task: Function) => Promise<void>;
+
+function createDebouncer(): Debouncer {
+  let queued = false;
+  return async (task: Function) => {
+    if (queued) {
+      return;
+    }
+    queued = true;
+    await Promise.resolve();
+    queued = false;
+    task();
+  };
+}
+
 export abstract class BaseElement extends HTMLElement {
+  private _debouncer: Debouncer;
+
   constructor() {
     super();
+    this._debouncer = createDebouncer();
   }
 
   connectedCallback() {
     this.renderCallback();
   }
 
   renderCallback() {
-    render(this.render(), this);
+    this._debouncer(() => {
+      render(this.render(), this);
```

Comment:
could you rename lit's render to `litRender`.

Reply:
Yes!